pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -906,3 +909,42 @@ type NomadSDConfig struct {
 	// +optional
 	EnableHTTP2 *bool `json:"enableHTTP2,omitempty"`
 }
+
+type LinodeSDConfig struct {
+	// BasicAuth information to use on every scrape request, required when role is robot.
+	// Role hcloud does not support basic auth.
```

Comment:
I'm confused about the role which isn't defined anywhere.
Also the Prometheus docs say:

> Optional HTTP basic authentication information, not currently supported by Linode APIv4.

Should we even have a field in the CRD for it then?

Reply:
Hmm.. The role comment is irrelevant indeed, I think I overlooked it while copying and pasting. 
About basic auth though, it is still part of the configuration. Should we keep it for that reason?

Reply:
It's included in the Prometheus configuration because it comes from the common HTTPClientConfig struct. But from what I can tell, Linode only supports token-based auth so it might be best to omit for now (and reduce the cognitive overhead).